great_expectations/datasource/data_connector/data_connector.py

```diff
@@ -196,7 +195,7 @@ def get_available_data_asset_names_and_types(self) -> List[Tuple[str, str]]:
 
     def get_batch_definition_list_from_batch_request(
         self,
-        batch_request: BatchRequest,
+        batch_request: BatchRequestBase,
```

Comment:
Should this be `Type[BatchRequestBase]`? Essentially a class that subclasses the base (but not the actual parent class)

Reply:
@cdkini I thought we wanted to subclass `BatchRequestBase` -- but not sure -- if you think Type is the best way, let us add it to a follow-on JIRA ticket to try.  Thanks.